src/raster/warp/create_and_reproject.rs

```diff
@@ -0,0 +1,250 @@
+use crate::errors::Result;
+use crate::raster::warp::GdalWarpOptions;
+use crate::spatial_ref::SpatialRef;
+use crate::utils::{_last_cpl_err, _path_to_c_string};
+use crate::{Dataset, DriverManager};
+use gdal_sys::CPLErr;
+use std::ffi::CString;
+use std::path::Path;
+use std::ptr;
+
+/// Settings for [`create_and_reproject`].
+#[derive(Debug, Clone, Default)]
+pub struct CreateReprojectOptions {
+    warp_options: GdalWarpOptions,
+    max_error: Option<f64>,
+    src_srs: Option<SpatialRef>,
+    src_nodata: Option<f64>,
+    dst_nodata: Option<f64>,
+    output_format: Option<String>,
+}
+
+impl CreateReprojectOptions {
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    /// Set the maximum error.
+    ///
+    /// Measured in input pixels, it is the allowed in approximating
+    /// transformations.
+    ///
+    /// `0.0` indicates exact calculations.
+    pub fn with_max_error(&mut self, max_error: f64) -> &mut Self {
+        self.max_error = Some(max_error);
+        self
+    }
+
+    /// Fetch the specified maximum error.
+    ///
+    /// Returns `None` if unset.
+    pub fn max_error(&self) -> Option<f64> {
+        self.max_error
+    }
+
+    /// Specify the source projection.
+    ///
+    /// If unset, the source projection is read from the source dataset.
+    /// If set, the source projection os overridden.
+    pub fn with_src_projection(&mut self, srs: &SpatialRef) -> &mut Self {
```

Comment:
- we should take `srs` by value instead of doing `srs.clone()`
 - the comment talks about "unset", but there's no `Option` around it
 - I think we use `projection` for strings and `spatial_ref` or `spatial_reference` for `SpatialRef`s.